### PR TITLE
Add content bucket DB metrics and reduce memory footprint

### PR DESCRIPTION
**Pull request by vekterli**
@geirst please review.

The 2nd commit reduces the initial memory footprint of a content node B-tree bucket DB by three orders of magnitude and then some 🙂 This adds up since there are always two of them present (global/non-global).

### CODE REVIEW

File: storage/src/vespa/storage/bucketdb/lockablemap.hpp [hunk 1]
```diff
@@ -85,6 +85,15 @@ LockableMap<Map>::getMemoryUsage() const noexcept
         sizeof(std::mutex) + sizeof(std::condition_variable);
 }
 
+template <typename Map>
+vespalib::MemoryUsage LockableMap<Map>::detailed_memory_usage() const noexcept {
+    // We don't have any details for this map type, just count everything as "allocated".
+    size_t used = getMemoryUsage();
+    vespalib::MemoryUsage mem_usage;
+    mem_usage.incAllocatedBytes(used);
```
Comment (geirst) [inline storage/src/vespa/storage/bucketdb/lockablemap.hpp:93]: You should probably count everything as 'used' as well.

**Review by geirst — APPROVED** [review]
👍

File: storage/src/vespa/storage/bucketdb/lockablemap.hpp (hunk 1 above)
Comment (vekterli) [reply storage/src/vespa/storage/bucketdb/lockablemap.hpp:93]: Good point, done

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**